ci/scripts/python_test.sh

```diff
@@ -54,4 +55,11 @@ export PYARROW_TEST_ORC
 export PYARROW_TEST_PARQUET
 export PYARROW_TEST_S3
 
+# Testing Arrow Python
+if [ "${ARROW_BUILD_TESTS}" == "ON" ]; then
+  pushd ${test_dir}
+  ctest
```

Comment:
```suggestion
  ctest --output-on-failure --timeout 300
```

I want to add `--parallel ${n_jobs}` too like `ci/scripts/cpp_test.sh`.

Reply:
Sure, will add 👍